### ISSUE TITLE
A question about spectral denoising

### ISSUE THREAD

**Issue by 13305912260** [conor-horgan/DeepeR#6]
Dear Dr. Horgan:

Good morning!

I have read your paper High-throughput molecular imaging via deep learning enabled Raman spectroscopy and your code[GitHub - conor-horgan/DeepeR: DeepeR: deep learning enabled Raman spectroscopy](https://github.com/conorhorgan/DeepeR).
They are of great help for my current research,

  I have reviewed the paper and code, and I would like to know how you specifically processed the 11 training data in the spectral denoising section? Do I need to shuffle the spectral pairs in every. mat file? Or perhaps there are other operations (I know I have done background removal and normalization).

Thank you for your answer.

Regards!
YANG CHEN

**Comment by conor-horgan** [comment]
Hi Yang,

Please see [here](https://pubs.acs.org/doi/suppl/10.1021/acs.analchem.1c02178/suppl_file/ac1c02178_si_001.pdf) for detailed materials and methods. 

Spectral preprocessing was applied as described here:
"Preliminary hyperspectral Raman image processing was kept constant for all datasets and performed using
Witec ProjectFOUR software. Spectra were cropped to the fingerprint spectral region (600 – 1800 cm-1) and
background autofluorescence subtraction performed using ProjectFOUR’s ‘shape’ background filter with α =
500. Hyperspectral images were then exported to MATLAB. Where applicable, Savitzky-Golay filtering (with
varying order and window length) was performed in Python using SciPy’s savgol_filter function, PCA
denoising was performed in Python using scikit-learn’s PCA function, and wavelet denoising was performed
in MATLAB using the wdenoise function in MATLAB’s Wavelet Toolbox. For PCA and wavelet denoising,
parameterisation was empirically determined, with selected parameters minimising the mean squared error
between the denoised spectra and the target high SNR spectra."

11-fold was cross-validation was performed as described here:
"Denoising of Raman spectra was achieved via a one-dimensional (1D) ResUNet architecture ([Figure 1](https://pubs.acs.org/doi/full/10.1021/acs.analchem.1c02178#fig1)). The network was trained for 500 epochs using the Adam optimizer, [(38)](javascript:void(0);) with an L1-norm loss function and a one-cycle learning rate scheduler. Eleven independent models were trained, with the Raman spectra from a single hyperspectral Raman cell image used as the test set in each case, while the training and validation sets were formed from the Raman spectra of the remaining 10 hyperspectral Raman cell images. Evaluation on the validation set was used to prevent overfitting, while all results presented are the mean across the 11 test set folds."

If you would like to re-perform cross-validation for training purposes, you can shuffle the spectral pairs in the .mat files as you suggest and split into N groups. Then you can train N models, where for each model a different group from the N groups is used as the test dataset.

**Comment by 13305912260** [comment]
Dear Dr. Horgan:
Thank you for your answer. It answered my question and I wish you all the best.

**Comment by 13305912260** [comment]
Hello, Dr. Horgan.

Today I tried your Raman super-resolution work, but after downloading the dataset, I noticed that the dimensions of these cells do not seem to be 64X64X500 as mentioned in the paper (ranging from as small as 40X40X500 to as large as 130X150X500). Is there anything I did wrong? Thank you for your answer
